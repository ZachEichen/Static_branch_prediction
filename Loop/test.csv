raw_string, loop_depth, number_BB, number_exits, number_exit_blocks, num_successors, isexit, isbackedge, isdestinationinloop, isdestinationnestedloop
"  br i1 %13, label %14, label %48, !prof !35",1, 11, 1, 1, 2, 1, 0, 1, 0, 53, 32, 32, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
"  br i1 %20, label %21, label %23, !prof !36",1, 11, 1, 1, 2, 0, 0, 1, 0, 53, 32, 20, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0
"  br i1 %28, label %29, label %43, !prof !37",2, 5, 1, 1, 2, 1, 0, 1, 0, 53, 32, 32, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
"  br i1 %33, label %34, label %38, !prof !38",2, 5, 1, 1, 2, 0, 0, 1, 0, 53, 23, 0, 0, 0, 0, 1, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
