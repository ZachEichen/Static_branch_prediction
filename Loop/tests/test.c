#include <stdio.h>

int main() {
    
    int a = 0;
    int b = 5;
    int c = 3;
    int d = 8;

    int i = 0;

    for(i = a; i<=b; i++){

        int j = 0;

        if(i==(a+b)/2) continue;

        for(j = c; j < d; j++){

            // Do stuff here
            printf("Hello World!");

            if(j % 2 == 0){
                printf("This is an even number.");
            }
        }
        
    }

    return 0;
}