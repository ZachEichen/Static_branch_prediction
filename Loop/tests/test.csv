raw_string, loop_depth, number_BB, number_exits, number_exit_blocks, num_successors, isexit, isbackedge, isdestinationinloop, isdestinationnestedloop, opcode_condition, prev_instr_1, prev_instr_2, operand_1.constant, operand_1.isfunctionarg, operand_1.isglobalvar, operand_1.isgeneralvar, operand_1.other, operand_2.constant, operand_2.isfunctionarg, operand_2.isglobalvar, operand_2.isgeneralvar, operand_2.other, operand_3.constant, operand_3.isfunctionarg, operand_3.isglobalvar, operand_3.isgeneralvar, operand_3.other, operand_4.constant, operand_4.isfunctionarg, operand_4.isglobalvar, operand_4.isgeneralvar, operand_4.other
"  br i1 %13, label %14, label %48, !prof !35",1, 11, 1, 1, 2, 1, 0, 1, 0, 53, 32, 32, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
"  br i1 %20, label %21, label %23, !prof !36",1, 11, 1, 1, 2, 0, 0, 1, 0, 53, 32, 20, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0
"  br i1 %28, label %29, label %43, !prof !37",2, 5, 1, 1, 2, 1, 0, 1, 0, 53, 32, 32, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
"  br i1 %33, label %34, label %38, !prof !38",2, 5, 1, 1, 2, 0, 0, 1, 0, 53, 23, 0, 0, 0, 0, 1, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
