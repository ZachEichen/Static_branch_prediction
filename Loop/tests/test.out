using build/pass/LoopAnalysisPass.dylib
This branch instruction does not have a loop associated with it.
Instruction: 
  br i1 %13, label %14, label %47, !prof !35
Loop Depth: 1
Number of BBs: 11
Number of Exits: 1
NUmber of Exit Blocks: 1
NUmber of Successors: 2
Is Exit?:  Yes 
Is Back Edge?:  No 
Is the Destination In Loop?:  Yes 
Is the Destination In Nested Loop?:  No 


Instruction: 
  br i1 %20, label %21, label %23, !prof !36
Loop Depth: 1
Number of BBs: 11
Number of Exits: 1
NUmber of Exit Blocks: 1
NUmber of Successors: 2
Is Exit?:  No 
Is Back Edge?:  No 
Is the Destination In Loop?:  Yes 
Is the Destination In Nested Loop?:  No 


Instruction: 
  br label %44
Loop Depth: 1
Number of BBs: 11
Number of Exits: 1
NUmber of Exit Blocks: 1
NUmber of Successors: 1
Is Exit?:  No 
Is Back Edge?:  No 
Is the Destination In Loop?:  Yes 
Is the Destination In Nested Loop?:  No 


Instruction: 
  br label %25
Loop Depth: 1
Number of BBs: 11
Number of Exits: 1
NUmber of Exit Blocks: 1
NUmber of Successors: 1
Is Exit?:  No 
Is Back Edge?:  No 
Is the Destination In Loop?:  Yes 
Is the Destination In Nested Loop?:  Yes 


Instruction: 
  br i1 %28, label %29, label %42, !prof !37
Loop Depth: 2
Number of BBs: 5
Number of Exits: 1
NUmber of Exit Blocks: 1
NUmber of Successors: 2
Is Exit?:  Yes 
Is Back Edge?:  No 
Is the Destination In Loop?:  Yes 
Is the Destination In Nested Loop?:  No 


Instruction: 
  br i1 %33, label %34, label %37, !prof !38
Loop Depth: 2
Number of BBs: 5
Number of Exits: 1
NUmber of Exit Blocks: 1
NUmber of Successors: 2
Is Exit?:  No 
Is Back Edge?:  No 
Is the Destination In Loop?:  Yes 
Is the Destination In Nested Loop?:  No 


Instruction: 
  br label %37
Loop Depth: 2
Number of BBs: 5
Number of Exits: 1
NUmber of Exit Blocks: 1
NUmber of Successors: 1
Is Exit?:  No 
Is Back Edge?:  No 
Is the Destination In Loop?:  Yes 
Is the Destination In Nested Loop?:  No 


Instruction: 
  br label %38
Loop Depth: 2
Number of BBs: 5
Number of Exits: 1
NUmber of Exit Blocks: 1
NUmber of Successors: 1
Is Exit?:  No 
Is Back Edge?:  No 
Is the Destination In Loop?:  Yes 
Is the Destination In Nested Loop?:  No 


Instruction: 
  br label %25, !llvm.loop !39
Loop Depth: 2
Number of BBs: 5
Number of Exits: 1
NUmber of Exit Blocks: 1
NUmber of Successors: 1
Is Exit?:  No 
Is Back Edge?:  Yes 
Is the Destination In Loop?:  Yes 
Is the Destination In Nested Loop?:  No 


Instruction: 
  br label %44
Loop Depth: 1
Number of BBs: 11
Number of Exits: 1
NUmber of Exit Blocks: 1
NUmber of Successors: 1
Is Exit?:  No 
Is Back Edge?:  No 
Is the Destination In Loop?:  Yes 
Is the Destination In Nested Loop?:  No 


Instruction: 
  br label %10, !llvm.loop !41
Loop Depth: 1
Number of BBs: 11
Number of Exits: 1
NUmber of Exit Blocks: 1
NUmber of Successors: 1
Is Exit?:  No 
Is Back Edge?:  Yes 
Is the Destination In Loop?:  Yes 
Is the Destination In Nested Loop?:  No 


